extensions/qxsck/var-and-list.js

```diff
@@ -5,292 +5,872 @@
+            if (arr[i] < arr[i + 1]) return false;
+          }
+          return true;
+        }
+      };
+    }
+
     getInfo() {
       return {
         id: "qxsckvarandlist",
-        name: Scratch.translate({ id: "name", default: "Variable and list" }),
-        color1: "#FF661A",
-        color2: "#EE6521",
+        name: Scratch.translate("Variable and list"),
+        color1: "#ed6b00",
+        color2: "#ed6b00",
+        blockIconURI: qxsckvarandlistIcon,
+        menuIconURI: qxsckvarandlistIcon,
         blocks: [
+          //command,reporter,Boolean,hat,conditional,loop
           {
-            opcode: "getVar",
-            blockType: Scratch.BlockType.REPORTER,
-            text: Scratch.translate({
-              id: "getVar",
-              default: "value of [VAR]",
-            }),
+            opcode: "haveVar",
+            blockType: "Boolean",
+            text: Scratch.translate("have variable [VAR] ?"),
             arguments: {
               VAR: {
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "variable",
               },
             },
-            extensions: ["colours_data"],
           },
           {
-            opcode: "seriVarsToJson",
-            blockType: Scratch.BlockType.REPORTER,
-            text: Scratch.translate({
-              id: "seriVarsToJson",
-              default: "convert all variables starting with [START] to json",
-            }),
+            opcode: "getVar",
+            blockType: "reporter",
+            text: Scratch.translate("value of variable [VAR]"),
             arguments: {
-              START: {
-                type: Scratch.ArgumentType.STRING,
+              VAR: {
+                type: "string",
                 defaultValue: "variable",
               },
             },
-            extensions: ["colours_data"],
           },
           {
             opcode: "setVar",
-            blockType: Scratch.BlockType.COMMAND,
-            text: Scratch.translate({
-              id: "setVar",
-              default: "set the value of [VAR] to [VALUE]",
-            }),
+            blockType: "command",
+            text: Scratch.translate("set variable [VAR] to [VALUE]"),
             arguments: {
               VAR: {
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "variable",
               },
               VALUE: {
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "value",
               },
             },
-            extensions: ["colours_data"],
+          },
+          {
+            opcode: "seriVarsToJson",
+            blockType: "reporter",
+            text: Scratch.translate(
+              "convert all variables starting with [START] to json"
+            ),
+            arguments: {
+              START: {
+                type: "string",
```

Comment:
Please use `Scratch.ArgumentType`. That's not a style thing, it's a compatibility thing.

Reply:
OK